pydantic/utils.py

```diff
@@ -787,3 +787,29 @@ def __setitem__(self, __key: Any, __value: Any) -> None:
         def __class_getitem__(cls, *args: Any) -> Any:
             # to avoid errors with 3.7
             pass
+
+
+URI_RESERVED_CHARACTERS = frozenset(":/?#[]@!$&'()*+,;=% ")
+
+
+def percent_encode(string: str, *, plus: bool = False) -> str:
+    """
+    Percent encode the URI reserved characters in a string
+    """
+    if not string:
+        return string
+
+    string = string.replace('%', f'%{ord("%"):02X}')
+
+    if plus:
+        string = string.replace('+', f'%{ord("+"):02X}')
+        string = string.replace(' ', '+')
+        characters = URI_RESERVED_CHARACTERS.symmetric_difference('+% ')
+    else:
+        characters = URI_RESERVED_CHARACTERS.symmetric_difference('%')
```

Comment:
surely we can just remove `%` from `URI_RESERVED_CHARACTERS` and remove this line.